theano/gof/graph.py

```diff
@@ -524,6 +526,11 @@ def eval(self, inputs_to_values=None):
     def __getstate__(self):
         d = self.__dict__.copy()
         d.pop("_fn_cache", None)
+        if (not config.pickle_test_value) and (hasattr(self.tag, 'test_value')):
+            warnings.warn("Test value of variable %s(%s) will not be dumped." % (d['auto_name'], d['name']))
```

Comment:
The code for the warning should check if the flag has the default value with the following ugly code:

`type(config).pickle_test_value.is_default`